sale_blanket_order_margin/models/blanket_order_line.py

```diff
@@ -0,0 +1,86 @@
+# Copyright 2022 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class BlanketOrderLine(models.Model):
+    _inherit = "sale.blanket.order.line"
+
+    margin = fields.Float(
+        compute="_compute_margin",
+        digits="Product Price",
+        store=True,
+        groups="base.group_user",
+    )
+    margin_percent = fields.Float(
+        "Margin (%)",
+        compute="_compute_margin",
+        store=True,
+        group_operator="avg",
+        groups="base.group_user",
+    )
+    purchase_price = fields.Float(
+        "Cost",
+        compute="_compute_purchase_price",
+        digits="Product Price",
+        store=True,
+        readonly=False,
```

Comment:
If the computed field is editable, don't you need to check if it was edited before triggering recomputation?